ctsit/nacculator
# Add Telephone Followup Packet module to nacculator

This branch introduces the final NACC packet, TFP (Telephone Followup Packet), to nacculator's conversion capabilities. The TFP is a UDS3 packet (like IVP and FVP) meant to replace any FVPs where the patient could not attend an in-person visit.

This change was made to expand nacculator's capabilities and account for all of the available NACC packets.


## Verification

_List the steps needed to make sure this thing works. Be precise._

1. Set up the proper virtual environment:
`source venv/bin/activate`
2. From the command-line, run:
`redcap2nacc -tfp <data.csv >data.txt`
with a valid tfp csv file in the nacculator directory. Test data can be found in our shared drive.
3. The output should be a properly-formatted .txt file with all PTIDs containing good data.
4. If you pass fields with data that should be blank, data outside of the allowed range, data with illegal characters, or fields with too many characters, the output should be an error listed in sys.err containing the details of the bad data.

- Verify the thing does this: creates a NACC-compatible TFP .txt file using data in a .csv file
- Verify the thing does not do that: does not give false positive or negative errors to the extent of nacculator's error-checking ability


## Affirmations

All of these should have a check by them. Any exception requires an explanation.

* [x] I have made an effort maintain a clear commit history (haven't merged other branches or rebased improperly). *note- a new version of nacculator was released while I was in the process of working on this branch. I merged the updated develop branch into this one to ensure there would be no conflicts.*
* [x] I have written the code myself or have given credit where credit is due. (most of builder.py and forms.py were generated by Cooper Martin shown in my commit history, and I made edits to that base code as needed)
